File: mouseInteraction.py
import cv2

def test(event, x, y, flags, param):
    if event == cv2.EVENT_LBUTTONDOWN:
        print("LBUTTON")

img1 = cv2.imread("./data/lena.jpg", -1)

cv2.namedWindow("image", flags=cv2.WINDOW_GUI_NORMAL)
cv2.imshow("image", img1)
cv2.setMouseCallback("image", test)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: mouseInteraction.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;

void mouseCallback(int event, int x, int y, int flags, void* userdata);

int main(int argc, char** argv){
	
	Mat image = imread("../data/lena.jpg", 1);

	imshow("Image", image);
	setMouseCallback("Image", mouseCallback, NULL);


	while(1){
		waitKey(0);
	}
	
	destroyAllWindows();
	return 0;
}

void mouseCallback(int event, int x, int y, int flags, void* userdata){
	if(event == EVENT_LBUTTONDOWN){
			std::cout << "Click" << std::endl;
	}	
}
